source/GameInterface/Services/MobileParties/Handlers/MobilePartyRecruitmentHandler.cs

```diff
@@ -0,0 +1,97 @@
+
+namespace GameInterface.Services.MobileParties.Handlers
+{
+    /// <summary>
+    /// Handles all mobile party recruitment in game
+    /// </summary>
+    public class MobilePartyRecruitmentHandler : IHandler
+    {
+        private readonly IMessageBroker messageBroker;
+        private readonly IObjectManager objectManager;
+        private readonly ILogger Logger = LogManager.GetLogger<MobilePartyRecruitmentHandler>();
+        private RecruitmentCampaignBehavior recruitmentCampaignBehavior;
+
+        public MobilePartyRecruitmentHandler(IMessageBroker messageBroker, IObjectManager objectManager)
+        {
+            this.messageBroker = messageBroker;
+            this.objectManager = objectManager;
+
+            messageBroker.Subscribe<UnitRecruitGranted>(Handle);
+            messageBroker.Subscribe<PartyRecruitGranted>(Handle);
+
+            
+        }
+
+        public void Dispose()
+        {
+            messageBroker.Unsubscribe<UnitRecruitGranted>(Handle);
+        }
+
+        //A Player Recruited Unit
+        private void Handle(MessagePayload<UnitRecruitGranted> obj)
+        {
+            recruitmentCampaignBehavior ??= Campaign.Current.CampaignBehaviorManager.GetBehavior<RecruitmentCampaignBehavior>();
+
+            var payload = obj.What;
+
+            if (objectManager.TryGetObject(payload.PartyId, out MobileParty mobileParty) == false)
+            {
+                Logger.Error("Could not handle SettlementEnterAllowed, PartyId not found: {id}", payload.PartyId);
+                return;
+            }
+
+            mobileParty.AddElementToMemberRoster(CharacterObject.Find(payload.CharacterId), payload.Amount);
+        }
+
+        private static readonly MethodInfo recruit_ApplyInternal = typeof(RecruitmentCampaignBehavior).GetMethod("ApplyInternal", BindingFlags.NonPublic | BindingFlags.Instance)!;
+        
+        //NPC Recruited Unit
+        private void Handle(MessagePayload<PartyRecruitGranted> obj)
+        {
+            var payload = obj.What;
+
+            if (objectManager.TryGetObject(payload.PartyId, out MobileParty mobileParty) == false)
+            {
+                Logger.Error("Could not handle {PartyRecruitGranted}, PartyId not found: {id}", nameof(PartyRecruitGranted), payload.PartyId);
+                return;
+            }
+            if (objectManager.TryGetObject(payload.SettlementId, out Settlement settlement) == false && payload.SettlementId != null)
+            {
+                Logger.Error("Could not handle {PartyRecruitGranted}, Settlement not found: {id}", nameof(PartyRecruitGranted), payload.SettlementId);
+                return;
+            }
+            if (objectManager.TryGetObject(payload.HeroId, out Hero hero) == false && payload.HeroId != null)
+            {
+                Logger.Error("Could not handle {PartyRecruitGranted}, HeroId not found: {id}", nameof(PartyRecruitGranted), payload.HeroId);
+                return;
+            }
```

Comment:
Should these ever be null?

```suggestion
            if (objectManager.TryGetObject(payload.SettlementId, out Settlement settlement) == false)
            {
                Logger.Error("Could not handle {PartyRecruitGranted}, Settlement not found: {id}", nameof(PartyRecruitGranted), payload.SettlementId);
                return;
            }
            if (objectManager.TryGetObject(payload.HeroId, out Hero hero) == false)
            {
                Logger.Error("Could not handle {PartyRecruitGranted}, HeroId not found: {id}", nameof(PartyRecruitGranted), payload.HeroId);
                return;
            }
```

Reply:
Yes, they are both usually null.